### PR TITLE
368 explain iks nodes new

### LINKED ISSUE #511: support IKS nodes in explainability
fixed by #509

### PR DESCRIPTION
**Pull request by ShiriMoran**
fixes #511

### CODE REVIEW

**Comment by ShiriMoran** [comment]
resolves #511

File: pkg/ibmvpc/examples/out/explain_out/vpeToIksNodeNoProtocolConn_all_vpcs_explain_debug.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Ingress: network ACL ky-test-private-2-others-acl allows connection; security group kube-clusterid:1 allows connection; security group ky-test-default-sg allows connection
+
+Path:
+	iks-clusterid:1[192.168.40.5] -> |
+
+
+Details:
+~~~~~~~~
+Egress:
+security group kube-r006-d7cfb31a-1d4b-40c8-83df-ce2e6f8f2e57 blocks connection since there are no relevant allow rules
```
Comment (adisos) [inline pkg/ibmvpc/examples/out/explain_out/vpeToIksNodeNoProtocolConn_all_vpcs_explain_debug.txt:16]: How is this SG `kube-r006-d7cfb31a-1d4b-40c8-83df-ce2e6f8f2e57` associated with `kube-clusterid:1` ? 
Shouldn't it be `kube-clusterid:1` ? 
The convention was that we identify the associated SG by the prefix "kube-" and then the cluster ID .
from the name of the iks node, it seems the cluster id was changed to `clusterid:1` ?
so I would expect the SG name to also be changed?
@olasaadi99  can you check?

File: pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	index: 4, direction: inbound,  conns: protocol: udp,  dstPorts: 30000-32767, cidr: 0.0.0.0/0
+	index: 6, direction: inbound,  conns: protocol: icmp,  icmpType: protocol: ICMP icmp-type: 8, cidr: 0.0.0.0/0
+	index: 7, direction: inbound, protocol: all, cidr: 192.168.0.4/32,192.168.4.4/32,192.168.8.4/32,192.168.16.4/32,192.168.20.4/32,192.168.24.4/32,192.168.32.4/32,192.168.36.4/32,192.168.40.4/32
+security group ky-test-default-sg allows connection with the following allow rules
+	index: 1, direction: inbound, protocol: all, cidr: 192.168.0.4/32,192.168.4.4/32,192.168.8.4/32,192.168.16.4/32,192.168.20.4/32,192.168.24.4/32,192.168.32.4/32,192.168.36.4/32,192.168.40.4/32
+	index: 2, direction: inbound, protocol: all, cidr: 0.0.0.0/0
+
+------------------------------------------------------------------------------------------------------------------------
+
+* Analysis of the connectivity of cluster worker nodes is under the assumption that the only security groups applied to them are the VPC default and the IKS generated SG
```
Comment (adisos) [inline pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt:35]: Do we want this message to be added also for connectivity output, not only explainability?

**Review by haim-kermany — COMMENTED** [review]
Code looks OK
One big general comment:
in our implementation, iks is IBM specific, and refer only at ibmvpc package.
however, you refer to it at vpcmodel.
did you check the option to implement in ibmvpc package?
@adisos, comment?

**Comment by adisos** [comment]
> Code looks OK One big general comment: in our implementation, iks is IBM specific, and refer only at ibmvpc package. however, you refer to it at vpcmodel. did you check the option to implement in ibmvpc package? @adisos, comment?

True, one option I can think of, is to expose from a Node interface a method that returns such relevant message string when analyzing this type of node, and have all the rest within ibmvpc pkg.

File: pkg/ibmvpc/examples/out/explain_out/vpeToIksNodeNoProtocolConn_all_vpcs_explain_debug.txt (hunk 1 above)
Comment (olasaadi99) [reply pkg/ibmvpc/examples/out/explain_out/vpeToIksNodeNoProtocolConn_all_vpcs_explain_debug.txt:16]: > How is this SG `kube-r006-d7cfb31a-1d4b-40c8-83df-ce2e6f8f2e57` associated with `kube-clusterid:1` ? Shouldn't it be `kube-clusterid:1` ? The convention was that we identify the associated SG by the prefix "kube-" and then the cluster ID . from the name of the iks node, it seems the cluster id was changed to `clusterid:1` ? so I would expect the SG name to also be changed? @olasaadi99 can you check?

I changed the pool Id with `clusterid:1`, `r006-d7cfb31a-1d4b-40c8-83df-ce2e6f8f2e57` was not relevant to my changes, commit `[e5bc95e](vscode-webview://073p1f54fp1ap9p4m3jh3qfp3entst322a8eic0j94brt6pj7eic/index.html?id=99634360-c035-479f-bcf6-74ceab34c8ea&origin=ca35c34c-9995-4395-a229-055190e1d29e&swVersion=4&extensionId=eamodio.gitlens&platform=electron&vscode-resource-base-authority=vscode-resource.vscode-cdn.net&parentOrigin=vscode-file%3A%2F%2Fvscode-app&purpose=webviewView#)`

File: pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt (hunk 2 above)
Comment (ShiriMoran) [reply pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt:35]: I think so.

**Comment by ShiriMoran** [comment]
> > Code looks OK One big general comment: in our implementation, iks is IBM specific, and refer only at ibmvpc package. however, you refer to it at vpcmodel. did you check the option to implement in ibmvpc package? @adisos, comment?
> 
> True, one option I can think of, is to expose from a Node interface a method that returns such relevant message string when analyzing this type of node, and have all the rest within ibmvpc pkg.

Which relevant message? "* Analysis of the connectivity of cluster worker..."? The message is per query, not sure I follow

**Comment by ShiriMoran** [comment]
> > Code looks OK One big general comment: in our implementation, iks is IBM specific, and refer only at ibmvpc package. however, you refer to it at vpcmodel. did you check the option to implement in ibmvpc package? @adisos, comment?
> 
> True, one option I can think of, is to expose from a Node interface a method that returns such relevant message string when analyzing this type of node, and have all the rest within ibmvpc pkg.

True. I'm aware of it - but what we need here is indeed specific to iksNode; The other option is to add a functionality that will be defined for all resources but used only for iksNode, which seemed even less appealing than using Kind() the way I did. Did you have anything else in mind @haim-kermany ?

File: pkg/ibmvpc/examples/out/explain_out/vpeToIksNodeNoProtocolConn_all_vpcs_explain_debug.txt (hunk 1 above)
Comment (ShiriMoran) [reply pkg/ibmvpc/examples/out/explain_out/vpeToIksNodeNoProtocolConn_all_vpcs_explain_debug.txt:16]: @adisos I asked @kyorav when verifying the new outputs the relevant node here is not `iksNode` but an `vpe`, and this is indeed the `SG` associated with it, as explicitly mentioned in `endpoint_gateways`

**Comment by haim-kermany** [comment]
> > > Code looks OK One big general comment: in our implementation, iks is IBM specific, and refer only at ibmvpc package. however, you refer to it at vpcmodel. did you check the option to implement in ibmvpc package? @adisos, comment?
> > 
> > 
> > True, one option I can think of, is to expose from a Node interface a method that returns such relevant message string when analyzing this type of node, and have all the rest within ibmvpc pkg.
> 
> True. I'm aware of it - but what we need here is indeed specific to iksNode; The other option is to add a functionality that will be defined for all resources but used only for iksNode, which seemed even less appealing than using Kind() the way I did. Did you have anything else in mind @haim-kermany ?

No, I didnt dive into the code to find another solution

File: pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt (hunk 2 above)
Comment (adisos) [reply pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt:35]: yes

Comment (ShiriMoran) [reply pkg/ibmvpc/examples/out/explain_out/IksNodeToIksNode_all_vpcs_explain_debug.txt:35]: #520

**Review by adisos — APPROVED** [review]

**Pull request merged**